File: app/src/main/java/za/co/woolworths/financial/services/android/ui/fragments/wreward/WRewardsLoggedOutFragment.java
package za.co.woolworths.financial.services.android.ui.fragments.wreward;

import android.animation.ObjectAnimator;
import android.content.Intent;
import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.RelativeLayout;
import android.widget.ScrollView;

import com.awfs.coordination.R;

import za.co.woolworths.financial.services.android.contracts.FirebaseManagerAnalyticsProperties;
import za.co.woolworths.financial.services.android.models.WoolworthsApplication;
import za.co.woolworths.financial.services.android.ui.activities.WRewardsMembersInfoActivity;
import za.co.woolworths.financial.services.android.ui.activities.dashboard.BottomNavigator;
import za.co.woolworths.financial.services.android.ui.views.WButton;
import za.co.woolworths.financial.services.android.ui.views.WTextView;
import za.co.woolworths.financial.services.android.util.ScreenManager;
import za.co.woolworths.financial.services.android.util.Utils;

/**
 * Created by W7099877 on 05/01/2017.
 */

public class WRewardsLoggedOutFragment extends Fragment implements View.OnClickListener {
	public WButton login;
	public WButton register;
	public RelativeLayout valuedMember;
	public RelativeLayout loyalMember;
	public RelativeLayout vipMember;
	public WTextView applyForWRewards;
	public WTextView wRewardsTagLine;
	private BottomNavigator mBottomNavigator;
	private ScrollView scrollLoggedOutLoggedIn;
	private String TAG = this.getClass().getSimpleName();

	@Nullable
	@Override
	public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
		View view = inflater.inflate(R.layout.wrewards_loggedout_loggedin_notlinked, container, false);
		try {
			mBottomNavigator = (BottomNavigator) getActivity();
			mBottomNavigator.removeToolbar();
		} catch (IllegalStateException ex) {
			Log.d(TAG, ex.toString());
		}
		view.findViewById(R.id.wRewars_linkaccounts).setVisibility(View.GONE);
		login = view.findViewById(R.id.wRewars_login);
		register = view.findViewById(R.id.wRewars_register);
		valuedMember = view.findViewById(R.id.layoutValuedMember);
		loyalMember = view.findViewById(R.id.layoutLoyalMember);
		vipMember = view.findViewById(R.id.layoutVipMember);
		applyForWRewards = view.findViewById(R.id.applyForWRewards);
		wRewardsTagLine = view.findViewById(R.id.wRewards_tag_line);
		scrollLoggedOutLoggedIn = view.findViewById(R.id.scrollLoggedOutLoggedIn);
		wRewardsTagLine.setText(getResources().getText(R.string.wrewards_logout_tag_line));
		login.setOnClickListener(this);
		register.setOnClickListener(this);
		valuedMember.setOnClickListener(this);
		loyalMember.setOnClickListener(this);
		vipMember.setOnClickListener(this);
		applyForWRewards.setOnClickListener(this);

		return view;
	}

	@Override
	public void onResume() {
		super.onResume();
		Utils.setScreenName(getActivity(), FirebaseManagerAnalyticsProperties.ScreenNames.WREWARDS_SIGNED_OUT);
	}

	@Override
	public void onClick(View v) {
		switch (v.getId()) {
			case R.id.wRewars_login:
				Utils.triggerFireBaseEvents(FirebaseManagerAnalyticsProperties.WREWARDSSIGNIN);
				ScreenManager.presentSSOSignin(getActivity());
				break;
			case R.id.wRewars_register:
				Utils.triggerFireBaseEvents(FirebaseManagerAnalyticsProperties.WREWARDSREGISTER);
				ScreenManager.presentSSORegister(getActivity());
				break;
			case R.id.layoutValuedMember:
				redirectToWRewardsMemberActivity(0);
				break;
			case R.id.layoutLoyalMember:
				redirectToWRewardsMemberActivity(1);
				break;
			case R.id.layoutVipMember:
				redirectToWRewardsMemberActivity(2);
				break;
			case R.id.applyForWRewards:
				Utils.openExternalLink(getActivity(), WoolworthsApplication.getWrewardsLink());
				break;
			default:
				break;
		}
	}

	@Override
	public void onActivityResult(int requestCode, int resultCode, Intent data) {
		super.onActivityResult(requestCode, resultCode, data);
	}

	public void redirectToWRewardsMemberActivity(int type) {
		startActivity(new Intent(getActivity(), WRewardsMembersInfoActivity.class).putExtra("type", type));
		getActivity().overridePendingTransition(R.anim.slide_up_anim, R.anim.stay);
	}

	@Override
	public void onHiddenChanged(boolean hidden) {
		super.onHiddenChanged(hidden);
		if (mBottomNavigator != null)
			mBottomNavigator.removeToolbar();
	}

	public void scrollToTop() {
		ObjectAnimator anim = ObjectAnimator.ofInt(scrollLoggedOutLoggedIn, "scrollY", scrollLoggedOutLoggedIn.getScrollY(), 0);
		anim.setDuration(500).start();
	}
}

File: app/src/main/java/za/co/woolworths/financial/services/android/ui/fragments/wreward/WRewardsLoggedOutFragment.kt
package za.co.woolworths.financial.services.android.ui.fragments.wreward

import android.animation.ObjectAnimator
import android.os.Bundle
import android.view.View
import com.awfs.coordination.R
import com.awfs.coordination.databinding.WrewardLogoutFragmentBinding
import za.co.woolworths.financial.services.android.contracts.FirebaseManagerAnalyticsProperties
import za.co.woolworths.financial.services.android.models.AppConfigSingleton
import za.co.woolworths.financial.services.android.ui.activities.dashboard.BottomNavigator
import za.co.woolworths.financial.services.android.ui.fragments.wreward.unique_locators.WRewardUniqueLocatorsHelper
import za.co.woolworths.financial.services.android.util.ScreenManager
import za.co.woolworths.financial.services.android.util.Utils

class WRewardsLoggedOutFragment : WRewardOnBoardingFragment(R.layout.wreward_logout_fragment), View.OnClickListener {

    private lateinit var binding: WrewardLogoutFragmentBinding
    private var mBottomNavigator: BottomNavigator? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mBottomNavigator = activity as? BottomNavigator?
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding = WrewardLogoutFragmentBinding.bind(view)

        binding.apply {
            mBottomNavigator?.removeToolbar()

            underlineText(incSignOutContent.tvSignIn)
            underlineText(incSignOutContent.tvRegister)

            setUpPager(vpJoinRewardInfo, tabIndicator)

            incSignOutContent.applyForWRewards.setOnClickListener(this@WRewardsLoggedOutFragment)
            incSignOutContent.tvSignIn.setOnClickListener(this@WRewardsLoggedOutFragment)
            incSignOutContent.tvRegister.setOnClickListener(this@WRewardsLoggedOutFragment)
            WRewardUniqueLocatorsHelper.setRewardsSignedOutMainIDs(incSignOutContent.applyForWRewards,incSignOutContent.tvSignIn,incSignOutContent.tvOr,incSignOutContent.tvRegister)
        }
    }

    override fun onClick(view: View?) {
        activity?.apply {
            when (view?.id) {
                R.id.applyForWRewards -> Utils.openLinkInInternalWebView( AppConfigSingleton.wrewardsLink, true)
                R.id.tvSignIn -> {
                    Utils.triggerFireBaseEvents(FirebaseManagerAnalyticsProperties.WREWARDSSIGNIN, this)
                    ScreenManager.presentSSOSignin(this)
                }
                R.id.tvRegister -> {
                    Utils.triggerFireBaseEvents(FirebaseManagerAnalyticsProperties.WREWARDSREGISTER, this)
                    ScreenManager.presentSSORegister(this)
                }
            }
        }
    }

    override fun onResume() {
        super.onResume()
        activity?.let { activity -> Utils.setScreenName(activity, FirebaseManagerAnalyticsProperties.ScreenNames.WREWARDS_SIGNED_OUT) }
    }

    fun scrollToTop() = binding.scrollLoggedOutLoggedIn?.let { view -> ObjectAnimator.ofInt(view, "scrollY", view.scrollY, 0).setDuration(500).start() }

    override fun onHiddenChanged(hidden: Boolean) {
        super.onHiddenChanged(hidden)
        mBottomNavigator?.removeToolbar()
    }
}
